core/src/main/java/org/jenkins/ui/icon/IconSet.java

```diff
@@ -57,6 +69,50 @@ public static void initPageVariables(JellyContext context) {
         context.setVariable("icons", icons);
     }
 
+    private String prependTitleIfRequired(String icon, String title) {
+        if (StringUtils.isNotBlank(title)) {
+            return "<span class=\"jenkins-visually-hidden\">" + title + "</span>" + icon;
+        }
+        return icon;
+    }
+
+    public String getIonicon(String name, String title) {
+        if (IONICONS.containsKey(name)) {
+            String icon = IONICONS.get(name);
+            return prependTitleIfRequired(icon, title);
+        }
+
+        // Load icon if it exists
+        InputStream inputStream = getClass().getResourceAsStream("/images/ionicons/" + name + ".svg");
+        String ionicon = null;
+
+        try {
+            if (inputStream != null) {
+                ionicon = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+            } else {
+                // this path is used in jetty:run
+                Path path = Paths.get("war/src/main/webapp/images/ionicons/" + name + ".svg");
+                boolean exists = Files.exists(path);
+                // try working directory in war folder
+                path = exists ? path : Paths.get("src/main/webapp/images/ionicons/" + name + ".svg");
+                ionicon = new String(Files.readAllBytes(path), StandardCharsets.UTF_8);
```

Comment:
Not sure if there's a better way to load a resource when running with `jetty:run` but they don't appear to be available through the regular classpath location, these two paths are based on running from either the root directory of a jenkins checkout or the war sub folder

Reply:
I'll try moving them to src/main/resources